File: backend/app/schemas/conversation.py
from typing import Optional, Literal, List
from datetime import datetime
from pydantic import BaseModel, Field
import json

RoleType = Literal["user", "assistant", "system"]


class ConversationCreateRequest(BaseModel):
    title: str = Field("新对话", description="会话标题")
    agent_id: int
    memory_data: Optional[dict] = None  # 新增的短期记忆数据字段


class ConversationCreateResponse(BaseModel):
    conversation_id: int


class ConversationListRequest(BaseModel):
    agent_id: int
    limit: int = 20
    offset: int = 0


class ConversationItem(BaseModel):
    id: int
    agent_id: int
    title: Optional[str]
    memory_data: Optional[dict] = None  # 新增的短期记忆数据字段
    created_at: datetime
    updated_at: datetime


class ConversationListResponse(BaseModel):
    conversations: List[ConversationItem]
    total: int


class MessageItem(BaseModel):
    id: int
    role: RoleType
    content: str
    created_at: datetime


class MessageListRequest(BaseModel):
    conversation_id: int
    limit: int = 50


class MessageListResponse(BaseModel):
    conversation_id: int
    total: int = 50
    messages: List[MessageItem]

File: frontend/src/types/conversation.ts
export interface ConversationCreateRequest {
  title?: string;
  agent_id: number;
}

export interface ConversationCreateResponse {
  conversation_id: number;
}

export interface ConversationListRequest {
  agent_id: number;
  limit?: number;
  offset?: number;
}

export interface ConversationItem {
  id: number;
  agent_id: number;
  title?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ConversationListResponse {
  conversations: ConversationItem[];
  total: number;
}
